### PR TITLE
:sparkles: Add --user option

### PR DESCRIPTION
**Pull request by mdengler**
This allows passing `--user` to `pip install`:

```
$ pipgrip --verbose --install --user jupyterlab-black
jupyterlab-black==0.2.1
[...]
Installing collected packages: attrs, backcall, bleach, decorator, zipp, importlib-metadata, ipython-genutils, traitlets, tornado, wcwidth, prompt-toolkit, pexpect, parso, jedi, pygments, setuptools, pickleshare, ipython, ipykernel, jinja2, mistune, nbformat, pandocfilters, prometheus-client, pyparsing, python-dateutil, pyzmq, send2trash, terminado, testpath
Successfully installed attrs-19.3.0 backcall-0.2.0 bleach-3.1.5 decorator-4.4.2 importlib-metadata-1.7.0 ipykernel-5.3.0 ipython-7.16.1 ipython-genutils-0.2.0 jedi-0.17.1 jinja2-2.11.2 mistune-0.8.4 nbformat-5.0.7 pandocfilters-1.4.2 parso-0.7.0 pexpect-4.8.0 pickleshare-0.7.5 prometheus-client-0.8.0 prompt-toolkit-3.0.5 pygments-2.6.1 pyparsing-2.4.7 python-dateutil-2.8.1 pyzmq-19.0.1 send2trash-1.5.0 setuptools-47.3.1 terminado-0.8.3 testpath-0.4.4 tornado-6.0.4 traitlets-4.3.3 wcwidth-0.2.5 zipp-3.1.0
$
```


Without this change:

```
$ pipgrip --verbose --install jupyterlab-black
[...]
Successfully built pandocfilters tornado
Installing collected packages: attrs, backcall, bleach, decorator, zipp, importlib-metadata, tornado, ipython-genutils, traitlets, pickleshare, pygments, parso, jedi, wcwidth, prompt-toolkit, pexpect, setuptools, ipython, ipykernel, jinja2, mistune, nbformat, pandocfilters, prometheus-client, pyparsing, python-dateutil, pyzmq, send2trash, terminado, testpath                                                                  ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/local/lib/python3.7'                        Consider using the `--user` option or check the permissions.                                                                                                                                                                                                                                                                                                                                                                              Error: Command '['/usr/bin/python', '-m', 'pip', 'install', '--cache-dir', '/home/martin/.cache/pip/wheels/pipgrip', '--index-url', 'https://pypi.org/simple', '--trusted-host', 'pypi.org', '--progress-bar=off', 'jupyterlab-black', '--constraint', '/tmp/tmp3vw71jhq']' returned non-zero exit status 1.                                                                                                                              $
```

### CODE REVIEW

File: src/pipgrip/pipper.py [hunk 1]
```diff
@@ -157,11 +158,18 @@ def _get_wheel_args(index_url, extra_index_url, pre, cache_dir=None):
...
 def install_packages(
-    packages, index_url, extra_index_url, pre, cache_dir, editable, constraints=None
+    packages,
+    index_url,
+    extra_index_url,
+    pre,
+    cache_dir,
+    editable,
+    constraints=None,
+    user=None,
```
Comment (ddelange) [inline src/pipgrip/pipper.py:168]: could you put user above constraints without default? (required positional argument)

install_packages is i think only called once, from cli.py, and should always pass the boolean (default False, maybe after [`--editable`](https://github.com/ddelange/pipgrip/blob/0.3.0/src/pipgrip/cli.py#L140)) and propagate it from there.

then afterwards `pipgrip -h` will give you updated output you can copypaste into the README :)

Comment (ddelange) [reply src/pipgrip/pipper.py:168]: a simple sanity check like for `editable` would also be nice: https://github.com/ddelange/pipgrip/blob/0.3.0/src/pipgrip/cli.py#L268-L270

Comment (mdengler) [reply src/pipgrip/pipper.py]: Thanks -- I think it's all done; perhaps you'll want to tweak the order of the arguments in `cli.py` or similar.

Comment (ddelange) [reply src/pipgrip/pipper.py]: Done! Now everywhere the order is install -- editable -- user. Also I kept `--user` only, to stay as close to the pip [interface](https://pip.pypa.io/en/stable/reference/pip_install/#usage) as possible (like with editable).

**Pull request merged**